src/main/java/dog/pawbook/logic/parser/ViewCommandParser.java

```diff
@@ -0,0 +1,37 @@
+package dog.pawbook.logic.parser;
+
+import static dog.pawbook.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+
+import dog.pawbook.logic.commands.ViewCommand;
+import dog.pawbook.logic.parser.exceptions.ParseException;
+
+/**
+ * Parses input arguments and creates a new FindCommand object
+ */
+public class ViewCommandParser implements Parser<ViewCommand> {
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the FindCommand
+     * and returns a FindCommand object for execution.
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    public ViewCommand parse(String args) throws ParseException {
+        String trimmedArgs = args.trim();
+
+        // Check if argument is empty
+        if (trimmedArgs.isEmpty()) {
+            throw new ParseException(
+                    String.format(MESSAGE_INVALID_COMMAND_FORMAT, ViewCommand.MESSAGE_USAGE));
+        }
+
+        // Check if string is an int
+        try {
+            Integer.parseInt(trimmedArgs);
+        } catch (NumberFormatException e) {
+            throw new ParseException(
+                    String.format(MESSAGE_INVALID_COMMAND_FORMAT, ViewCommand.MESSAGE_USAGE));
+        }
+
+        return new ViewCommand(Integer.parseInt(trimmedArgs));
```

Comment:
Maybe can declare an integer on top first but don't initialise it? then in the try part can use the result of parseInt such that it can be directly returned below.